### PR TITLE
Update wp.data.subscribe docs

### PR DESCRIPTION
**Pull request by Shelob9**
For: #5032
Related: #5003

## Description
This updates the docs for wp.data.subscribe for changes in #5003

## How Has This Been Tested?
* Pulled master, and ran the build script.
* Put code in a block.
* Saw it logged correctly.

- [x ] My code is tested.
- [x ] My code follows the WordPress code style.
- [x ] My code has proper inline documentation.

### CODE REVIEW

File: data/README.md [hunk 1]
```diff
@@ -71,12 +71,24 @@ Function used to subscribe to data changes. The listener function is called each
...
+		//Post title
+		title: wp.data.select( 'core/editor', 'getEditedPostAttribute', 'title' ),
+		//Post slug
+		slug: wp.data.select( 'core/editor', 'getEditedPostAttribute', 'slug' ),
+		//Post type
+		type: wp.data.select( 'core/editor', 'getEditedPostAttribute', 'type' ),
+		//Post content
+		content :wp.data.select( 'core/editor', 'getEditedPostAttribute', 'content' ),
+		//Number of blocks currently selected
+		selectedBlockCount: wp.data.select( 'core/editor', 'getSelectedBlockCount' ),
```
Comment (youknowriad) [inline data/README.md]: The Documentation of the data module is meant to be generic. Should we document these selectors as part of the `editor` module's documentation instead?

Also, in Calypso we have a script to generate selectors documentation automatically, we might be able to adapt it here cc @aduth

**Comment by aduth** [comment]
The API for `select` has been revised as of #5007.

Also, you may be encountering errors because tests were briefly failing on master which was resolved in 6619a9f105063a0b6f57544f0760dc0aae69843e. You can remedy this by rebasing and pushing your branch:

```
# Assuming "upstream" is WordPress/Gutenberg
git checkout patch-2
git rebase upstream/master
git push -f origin patch-2
```

**Comment by aduth** [comment]
Any chance you'd be able to update this for the changes introduced by #5007?

Per Riad's feedback, I feel as though this should be limited to updating from:

```
getEditedPostSlug
```

To:

```
getEditedPostAttribute( 'slug' )
```

**Comment by danielbachhuber** [comment]
Closing re https://github.com/WordPress/gutenberg/issues/5032#issuecomment-377522394

**Pull request closed without merging**